kubetest2/internal/deployers/eksapi/k8s.go

```diff
@@ -110,3 +123,65 @@ func createAWSAuthConfigMap(client *kubernetes.Clientset, nodeNameStrategy strin
 	}, metav1.CreateOptions{})
 	return err
 }
+
+func emitNodeMetrics(metricRegistry metrics.MetricRegistry, k8sClient *kubernetes.Clientset, ec2Client *ec2.Client) error {
+	nodes, err := getReadyNodes(k8sClient)
+	if err != nil {
+		return err
+	}
+	var errs []error
+	for _, node := range nodes {
+		providerId, err := parseKubernetesProviderID(node.Spec.ProviderID)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		instanceInfo, err := ec2Client.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{
+			InstanceIds: []string{providerId.InstanceID},
+		})
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		launchTime := *instanceInfo.Reservations[0].Instances[0].LaunchTime
+		timeToRegistration := node.ObjectMeta.CreationTimestamp.Time.Sub(launchTime)
+		timeToReady := getNodeReadyCondition(&node).LastTransitionTime.Time.Sub(launchTime)
+
+		nodeDimensions := map[string]string{
+			"instanceType": node.ObjectMeta.Labels[corev1.LabelInstanceTypeStable],
```

Comment:
can we get the OS Image here from the node info or is that too specific?

Reply:
you mean the AMI ID or something else?

Reply:
maybe starting with like OS distro? i know the node exposes image info like `Amazon Linux 2023.4.20240611`

Reply:
gotcha, yeah that'd be useful. I remember seeing `AmazonLinux2` in kubectl output, so yeah it's gotta be somewhere. We can always call ec2:DescribeImages if not

Reply:
added, also pulled what I can from NodeInfo vs. labels